kong/llm/drivers/shared.lua

```diff
@@ -117,57 +165,67 @@ function _M.to_ollama(request_table, model)
 end
 
 function _M.from_ollama(response_string, model_info, route_type)
+  local output, _, analytics
+
   local response_table, err = cjson.decode(response_string)
   if err then
     return nil, "failed to decode ollama response"
   end
 
-  -- there is no direct field indicating STOP reason, so calculate it manually
-  local stop_length = (model_info.options and model_info.options.max_tokens) or -1
-  local stop_reason = "stop"
-  if response_table.eval_count and response_table.eval_count == stop_length then
-    stop_reason = "length"
-  end
+  if route_type == "stream/llm/v1/chat" then
+    output, _, analytics = handle_stream_event(response_table, model_info, route_type)
 
-  local output = {}
-
-  -- common fields
-  output.model = response_table.model
-  output.created = response_table.created_at
-
-  -- analytics
-  output.usage = {
-    completion_tokens = response_table.eval_count or 0,
-    prompt_tokens = response_table.prompt_eval_count or 0,
-    total_tokens = (response_table.eval_count or 0) + 
-                   (response_table.prompt_eval_count or 0),
-  }
-
-  if route_type == "llm/v1/chat" then
-    output.object = "chat.completion"
-    output.choices = {
-      [1] = {
-        finish_reason = stop_reason,
-        index = 0,
-        message = response_table.message,
-      }
+  elseif route_type == "stream/llm/v1/completions" then
+    output, _, analytics = handle_stream_event(response_table, model_info, route_type)
+
+  else
+    -- there is no direct field indicating STOP reason, so calculate it manually
+    local stop_length = (model_info.options and model_info.options.max_tokens) or -1
+    local stop_reason = "stop"
+    if response_table.eval_count and response_table.eval_count == stop_length then
+      stop_reason = "length"
+    end
+
+    output = {}
+
+    -- common fields
+    output.model = response_table.model
+    output.created = response_table.created_at
+
+    -- analytics
+    output.usage = {
+      completion_tokens = response_table.eval_count or 0,
+      prompt_tokens = response_table.prompt_eval_count or 0,
+      total_tokens = (response_table.eval_count or 0) + 
+                    (response_table.prompt_eval_count or 0),
     }
 
-  elseif route_type == "llm/v1/completions" then
-    output.object = "text_completion"
-    output.choices = {
-      [1] = {
-        index = 0,
-        text = response_table.response,
+    if route_type == "llm/v1/chat" then
+      output.object = "chat.completion"
+      output.choices = {
+        [1] = {
+          finish_reason = stop_reason,
+          index = 0,
+          message = response_table.message,
+        }
       }
-    }
 
-  else
-    return nil, "no ollama-format transformer for response type " .. route_type
+    elseif route_type == "llm/v1/completions" then
+      output.object = "text_completion"
+      output.choices = {
+        [1] = {
+          index = 0,
+          text = response_table.response,
+        }
+      }
```

Comment:
Dunno if I've mentioned it before on your PRs, but explicitly providing the numeric index for array-like tables is not actually required. So this:

```lua
      output.choices = {
        [1] = {
          index = 0,
          text = response_table.response,
        }
      }
```

Could be more tersely expressed this way:

```lua
      output.choices = {
        {
          index = 0,
          text = response_table.response,
        }
      }
```

Less of a change-request and more of an FYI as I think the code is perfectly legible as-is.

Reply:
Fixed in 2dd44a838d60c252f556366abc9bf6496a44129f